pkg/system/phase2_creating.go

```diff
@@ -92,8 +92,19 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	if err := r.ReconcileObject(r.ServiceMgmt, r.SetDesiredServiceMgmt); err != nil {
 		return err
 	}
-	if err := r.ReconcileObject(r.ServiceDb, r.SetDesiredServiceDB); err != nil {
-		return err
+	if r.NooBaa.Spec.DBType == "postgres" {
+		if err := r.ReconcileObject(r.ServiceDbPg, r.SetDesiredServiceDB); err != nil {
+			return err
+		}
+	} else {
+		if err := r.ReconcileObject(r.ServiceDb, r.SetDesiredServiceDB); err != nil {
+			return err
+		}
+	}
+	if r.NooBaa.Spec.DBType == "postgres" {
+		if err := r.UpgradeMigrateDB(); err != nil {
```

Comment:
that's good for now. I think that eventually we should have a more general upgrade phase, where we run an upgrade job that fits the context in which the operator operates in (to\from version, OCP\k8s, etc.)

Reply:
won't be changed for now